File: Derek Wei/warmups/week_01/raindrops/js/raindrops.js
// ## Warmup - Raindrops
//
// Write a Javascript program that will take a number (eg 28 or 1755 or 9, etc) and output the following:
//
// - If the number contains 3 as a factor, output 'Pling'.
// - If the number contains 5 as a factor, output 'Plang'.
// - If the number contains 7 as a factor, output 'Plong'.
// - If the number does not contain 3, 5 or 7 as a factor, output the number as a string.
//
//
// ### Examples
//
// - 28 has 7 as a factor.
//   - In raindrop-speak, this would be a simple "Plong".
// - 1755 has 3 and 5 as factors.
//   - In raindrop-speak, this would be a "PlingPlang".
// - 34 has neither 3, 5 nor 7 as a factor.
//   - Raindrop-speak doesn't know what to make of that, so it just goes with the straightforward "34".


const raindrops = (num) => {

  let result = "";

  if (num % 3 === 0) {
    result = result.concat("Pling");  // can replace with - result += "Pling"
  }
  if (num % 5 === 0) {
    result = result.concat("Plang");
  }
  if (num % 7 === 0) {
    result = result.concat("Plong");
  }
  if (!(num % 3 === 0 || num % 5 === 0 || num % 7 === 0)) {   // can replace with - if (result === "")
    result = num.toString();
  }

  console.log(result);
  return result;
}

raindrops(9); // contains 3 as a factor
raindrops(10); // contains 5 as a factor
raindrops(14); // contains 7 as a factor
raindrops(15); // contains 3 and 5 as factors
raindrops(21); // contains 3 and 7 as factors
raindrops(35); // contains 5 and 7 as factors
raindrops(105); // contains 3, 5 and 7 as factors
raindrops(34); // does not contain 3, 5 or 7 as factors

raindrops(10231241);
raindrops(1023213);
raindrops(8579947);
raindrops(2.5);

File: Tracey Hill/warmups/week_04/raindrops/raindrops.rb
require 'pry'
# Warmup - Raindrops

# Write a program using Ruby that will take a number (eg 28 or 1755 or 9, etc) and output the following:




#
# print ""



def pling_plang_plong(magic_number)
    magicResult = ''
    # If the number contains 3 as a factor, output 'Pling'.
    if magic_number % 3 == 0
        magicResult += 'Pling '
        puts magicResult
    end
    # If the number contains 5 as a factor, output 'Plang'.
    if magic_number % 5 == 0
        magicResult.to_s += 'Plang'
        puts magicResult
    end
    # If the number contains 7 as a factor, output 'Plong'.
    if magic_number % 7 == 0
            magicResult += 'Plong'
    end
    # If the number does not contain 3, 5, or 7 as a factor, output the number as a string.
    if magicResult == ''
            magicResult = number.to_s
    end
end

def menu
    puts "raindrop" * 25
    puts "Pling Plang Plong".center(50)
    puts

print "enter to start and q to quit"
end 

menu

choice = gets[0].downcase

until choice = 'q'
    print "enter a number"
    num = gets.to_i
    pling_plang_plong(magic_number)
end

puts 'Thanks for playing.'

# Examples

# 28 has 7 as a factor. In raindrop-speak, this would be a simple "Plong".
pling_plang_plong(28)
# 1755 has 3 and 5 as factors. In raindrop-speak, this would be a "PlingPlang".
pling_plang_plong(1755)
# 34 has neither 3, 5 nor 7 as a factor. Raindrop-speak doesn't know what to make of that, so it just goes with the straightforward "34".
pling_plang_plong(34)
binding.pry